kore/src/Kore/Internal/SideCondition.hs

```diff
@@ -414,11 +439,18 @@ fromDefinedTerms ::
 fromDefinedTerms definedTerms =
     top{definedTerms}
 
+{- | Prepares the 'SideCondition' for storing in the term attributes.
+ Any metadata information used only in particular places during execution
+ is erased.
+-}
 toRepresentation ::
     InternalVariable variable =>
     SideCondition variable ->
     SideCondition.Representation
-toRepresentation = mkRepresentation
+toRepresentation sideCondition =
```

Comment:
It occurs to me that in this function, we should also strip the `Simplified` attribute off every `Predicate` and `TermLike`, because those can refer back to earlier `SideCondition`s, which would be pretty inefficient. That's just a thought, not something to do in this pull request.

Reply:
Yeah, I remember this was actually used somehow, but I can't remember the details? Anyway, I'm sure it can be terrible for performance and it has also been terrible for debugging. I'll create an issue to remove it separately in case anything actually breaks, thanks for pointing this out!